realm/realm-library/src/androidTest/java/io/realm/DynamicRealmObjectTests.java

```diff
@@ -1629,4 +1631,95 @@ public void run() {
         thread.start();
         TestHelper.awaitOrFail(threadFinished);
     }
+
+    @Test
+    public void getNullableFields() {
+        realm.executeTransaction(realm -> {
+            NullablePrimitiveFields primitiveNullables = realm.createObject(NullablePrimitiveFields.class);
+            primitiveNullables.setFieldBoolean(null);
+
+            assertNull(primitiveNullables.getFieldBoolean());
```

Comment:
all of these should be tested below?

Reply:
Yes. I was just asserting that they were actually null when retrieved from the typed realm.